src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
I'm not seeing a need to modify the rules after the initial sort in `transform-code` is done.  Even if we add the ability to do a runtime reload of rules from disk later on, I think we'll be able to do that in such a way that the new rules can be sorted one time before replacing what is currently in the service context.  So, I'd like to pursue this change unless you see any major problems with it.